packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx

```diff
@@ -114,24 +114,26 @@ export class TooltipHost extends BaseComponent<ITooltipHostProps, ITooltipHostSt
       }
     }
 
-    this.setState({
-      isTooltipVisible: true
-    });
+    this.toggleTooltip(true);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipMouseLeave(ev: any) {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipCallOutDismiss() {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
+  }
+
+  private toggleTooltip(isTooltipVisible: boolean) {
+    if (this.props.onTooltipToggle && this.state.isTooltipVisible !== isTooltipVisible) {
+      this.props.onTooltipToggle(isTooltipVisible);
+    }
```

Comment:
At least we need to save previous state to compare later, when state has already changed.
And could be irrelevant, but given that setState is batched, we could get calls at the same time with opposite values.
As we don't need render to notify, I'm safer doing it apart.